libs/metrics-core/src/Data/Metrics.hs

```diff
@@ -99,13 +99,39 @@ metrics = liftIO $ Metrics
 --   This is to provide back compatibility with the previous collect-d metric names
 --   which often had paths and dot-separated names.
 --
--- Currently just replaces all "/" and "." with "_" and lowercases the result
+-- E.g. we sanitize a metric name like "net.resources._conversations_:cnv-members_:usr.DELETE.time.960"
+-- into: "net_resources_conversations_:cnv_members_:usr_delete_time_960"
 toInfo :: Path -> P.Info
-toInfo (Path p) = P.Info (p & T.replace "." "_"
-                            & T.replace "/" "_"
-                            & T.toLower)
-                        "description not provided"
-
+toInfo (Path p) =
+    P.Info (p
+            & T.map sanitize
+            & ensureValidStartingChar
+            & collapseMultipleUnderscores
+            & T.toLower)
+           "description not provided"
+  where
+    ensureValidStartingChar :: Text -> Text
+    ensureValidStartingChar = T.dropWhile (not . validStartingChar)
+
+    -- | prometheus metric names must start with an alphabetic char or a ':'
```

Comment:
`_` is also a valid starting char.

Reply:
Perhaps link to this page in a comment:
https://prometheus.io/docs/concepts/data_model/